exercise-app/src/main/java/org/exercise/app/AppController.java

```diff
@@ -0,0 +1,36 @@
+package org.exercise.app;
+
+import java.io.PrintStream;
+
+import com.examples.attsw.exercise.core.controller.IEmployeeController;
+
+public class AppController {
+
+	private IEmployeeController employeeController;
+
+	public AppController(IEmployeeController employeeController) {
+		this.employeeController = employeeController;
+	}
+
+	public void performAction(String actionCode, String arg, PrintStream out) throws IllegalArgumentException {
+		if (actionCode.equals("showAll") || actionCode.equals("showOne")) {
+			String allEmployees = employeeController.getAllEmployees();
```

Comment:
You call getAllEmployees also in the case of a single employee: it's a waste of resources.